tensorboard/webapp/runs/store/runs_types.ts

```diff
@@ -50,8 +50,8 @@ export interface RunsDataRoutefulState {
   defaultRunColorForGroupBy: Map<RunId, string>;
   runColorOverrideForGroupBy: Map<RunId, string>;
   groupKeyToColorString: Map<string, string>;
-  userSetGroupBy?: GroupBy;
-  initialGroupBy: GroupBy;
+  userSetGroupBy?: GroupByKey;
+  initialGroupBy: GroupByKey;
```

Comment:
This should not change to `GroupByKey` for the stronger typing reasons. It should remain as `GroupBy`. You should only change ` userSetGroupBy` to something like below:

```ts
userSetGroupByKey?: GroupByKey;
colorGroupRegexString?: string;
```